### DISCUSSION TITLE
hideOnClickOutside for the SideModal component

### DISCUSSION THREAD

**Discussion by pratapdd** [twilio-labs/paste#2810] [General] [answered]
SideModal documentation says consumers would be able to use the props of [Non-Modal Dialog Primitive documentation](https://paste.twilio.design/primitives/non-modal-dialog-primitive). And one of the property is `hideOnClickOutside`, but inside SideModal component at the moment it's hardcoded to false.  Is there any use case for the same or could we expect this would be available in future.

**Comment by SiTaggart** [comment, marked as answer]
Hi @pratapdd 

This looks like it might be a little bit of a misunderstanding from reading the docs. What we say in  the [using state hooks](https://paste.twilio.design/components/side-modal/#using-state-hooks) section is that you can use the hook, and state properties provided by it to control the side modal state. `hideOnClickOutside` is not part of the state hook arguments or returned state.

We're also fairly explicit in the Side modals intended behavior that it "does not block the user from interacting with the rest of the page", which if we allowed `hideOnClickOutside` to be overridden, would be circumvented. You can't interact with the underlying content on the page if every time you click outside, the dialog closes.

I wouldn't be expecting that functionality to be added to the component. If you do feel you really need it, you can use the non-modal dialog primitive directly to create your own custom component.
